### PR TITLE
Search: Migrate tests from Jetpack plugin into Search package

### LINKED ISSUE #23233: S1.7 Migrate Search PHP tests
Migrate the search PHP tests still in the Jetpack plugin into the package:

`projects/plugins/jetpack/tests/php/modules/search/test-class-jetpack-search.php`

### PR DESCRIPTION
**Pull request by jsnmoon**
Fixes #23233.

#### Changes proposed in this Pull Request:

Migrates tests from the Jetpack plugin into the Search package.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Ensure that the newly migrated tests pass (CI & `composer test-php` locally).

### CODE REVIEW

**Comment by bluefuton** [comment]
> Ensure that the newly migrated tests pass.

What's the command to run the tests? :)

**Comment by jsnmoon** [comment]
> > Ensure that the newly migrated tests pass.
> 
> What's the command to run the tests? :)

Updated the review instructions! Looks like I flubbed some of the test code during the migration, so I'm marking this In Progress until it's ready again.

File: projects/packages/search/composer.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
 	"require-dev": {
 		"automattic/jetpack-changelogger": "^3.0",
 		"yoast/phpunit-polyfills": "1.0.3",
-		"automattic/wordbless": "0.3.1"
+		"automattic/wordbless": "0.3.1",
+		"phpunit/phpunit": "^8.5.25"
```
Comment (kangzj) [inline projects/packages/search/composer.json:19]: 👍

File: projects/packages/search/composer.json [hunk 2]
```diff
@@ -46,6 +47,9 @@
 			"@composer phpunit"
 		],
 		"post-update-cmd": "php -r \"copy('vendor/automattic/wordbless/src/dbless-wpdb.php', 'wordpress/wp-content/db.php');\"",
+		"skip-test-php": [
```
Comment (kangzj) [inline projects/packages/search/composer.json:50]: Can I ask why we skip tests for lower PHP versions?

Comment (jsnmoon) [reply projects/packages/search/composer.json:50]: Ah, good question: PHPUnit versions 8.5.12 and above require PHP >= 7.2.0. I went ahead and added this in the code comments.

Comment (kangzj) [reply projects/packages/search/composer.json:50]: I'm a bit concerned if we'd possibly have some PHP compatibility issue not caught if we skip those, would we? I looked at a couple of other package/plugin, I think they just ignored PHP version requirement. If phpunit doesn't break  for lower PHP versions in our case. Is it better we still run the tests?

Comment (jsnmoon) [reply projects/packages/search/composer.json:50]: It does break for PHPUnit specified in the composer file; let me investigate downgrading our PHPUnit version as much as I can before our tests begin to fail.

Comment (jsnmoon) [reply projects/packages/search/composer.json]: Fixed!

Comment (jsnmoon) [reply projects/packages/search/composer.json]: Fixed!

**Review by kangzj — DISMISSED** [review]
Great work LGTM 👍

File: projects/packages/search/tests/php/test-jetpack-search.php [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
+class Test_Jetpack_Search extends TestCase {
 
 	/**
 	 * Verify deprecated classes still exist.
 	 *
 	 * @since 10.6.1
 	 */
 	public function test_deprecated_jetpack_search_class() {
-		$search = Jetpack_Search::instance();
+		$search = CLassic_Search::instance();
```
Comment (adimoldovan) [inline projects/packages/search/tests/php/test-jetpack-search.php:23]: Typo. Strange that it doesn't fail when running all the tests. 
It fails when running with a filter tough: `--filter=Test_Jetpack_Search`.

```suggestion
		$search = Classic_Search::instance();
```

Comment (jsnmoon) [reply projects/packages/search/tests/php/test-jetpack-search.php]: Ah, strange indeed. Fixed!

**Review by jeherve — DISMISSED** [review]

**Review by samiff — APPROVED** [review]

**Comment by jsnmoon** [comment]
🚢 r242555-wpcom

**Pull request merged**